src/test/RASTER_IMAGE_DATA_PERFORMANCE.test.ts

```diff
@@ -0,0 +1,383 @@
+import {CARTA} from "carta-protobuf";
+import * as Utility from "./testUtilityFunction";
+
+let WebSocket = require("ws");
+let testServerUrl = "ws://localhost:50505";
+let expectRootPath = "/Users/zarda/CARTA/Images";
+// let testSubdirectoryName = "QA"; // for NRAO backend
+let testSubdirectoryName = `${expectRootPath}/QA`; // ASIAA backend
+let connectionTimeout = 1000;
+let disconnectionTimeout = 1000;
+let openFileTimeout = 60000; // The larger file, the more required time.
+let readFileTimeout = 60000;
+let count: number[];
+
+describe("RASTER_IMAGE_DATA_PERFORMANCE tests", () => {   
+    let Connection: WebSocket;
+
+    beforeEach( done => {
+        // Establish a websocket connection in the binary form: arraybuffer 
+        Connection = new WebSocket(testServerUrl);
+        Connection.binaryType = "arraybuffer";
+        // While open a Websocket
+        Connection.onopen = () => {
+            // Checkout if Websocket server is ready
+            if (Connection.readyState === WebSocket.OPEN) {
+                // Preapare the message on a eventData
+                const message = CARTA.RegisterViewer.create({sessionId: "", apiKey: "1234"});
+                let payload = CARTA.RegisterViewer.encode(message).finish();
+                let eventData = new Uint8Array(32 + 4 + payload.byteLength);
+
+                eventData.set(Utility.stringToUint8Array("REGISTER_VIEWER", 32));
+                eventData.set(new Uint8Array(new Uint32Array([1]).buffer), 32);
+                eventData.set(payload, 36);
+
+                Connection.send(eventData);
+            } else {
+                console.log(`Can not open a connection.`);
+            }
+            done();
+        };
+    }, connectionTimeout);
+
+    test(`connect to CARTA "${testServerUrl}" & ...`, 
+    done => {
```

Comment:
again, not sure why we need to test these again each time.

Reply:
I think for each test image in this performance test, we either start a new carta session for each image, or we use a single session for all test images (this can test if backend resources are handled well or not). Probably we should do both?

Reply:
It's fine to start a new test each time, so each test will be independent. But we don't need to explicitly test that the file loads and file info is correct each time.

Reply:
I suppose to make a single test for one development at one purpose. That is after we make sure in which function the bug happens, the programer only needs to focus on a signle testing item. In the mean while, there is some chance to have other issue about connection and OS. 
And another reason is that there will be too many error messages involved each other at a test which has many assertions. My solution is to run every assertion at a renew process.

Reply:
it's not about renewing the process, that's fine, and that's a sensible idea, since each test will then be independent of the the previous test. However, when testing image performance, or spatial profiles, or any other feature, there's no need to re-test the file loading or file info. We already have separate tests for that.

Reply:
multiple assertions in a single test are also fine, because the test utility will specify which assertion failed.

Reply:
The main purpose of re-testing the file loading and file info is not to assert the programme codes. It may happen there is no supported meta data for loading a certain fits file for instance. I suppose this kind of testing will be removed to other testing about "support file format". I will change the code to not re-test file loading again.

Reply:
in terms of file support, that should be handled by a completely separate test. When the frontend sends a file open event to the backend, the backend will respond with an OPEN_FILE_ACK message, which will contain sufficient information to determine whether the file is supported or not. So there is no neat to combine that with tests of raster image data

Reply:
only left the assertion "file opens" to show a based result to compare the benchmark.

Reply:
PASS  src/test/CURSOR_Z_PROFILE_PERFORMANCE.test.ts (7.668s)
  CURSOR_Z_PROFILE_PERFORMANCE tests
    ✓ connect to CARTA "ws://localhost:50505" & ... (5ms)
    access directory
      ✓ assert the directory "/Users/zarda/CARTA/Images" opens. (8ms)
      ✓ assert the directory "/Users/zarda/CARTA/Images/QA" opens. (9ms)
    prepare the files
      ✓ assert the file "SgrB2-N.spw0.line.fits" opens. (46ms)
      ✓ assert a random cursor at round 1. (7117ms)
      ✓ assert a random cursor at round 2. (96ms)
      ✓ assert a random cursor at round 3. (94ms)
      ✓ assert a random cursor at round 4. (103ms)
      ✓ assert a random cursor at round 5. (100ms)

  console.log src/test/CURSOR_Z_PROFILE_PERFORMANCE.test.ts:258
    set cursor randomly: returning time = 7037,6,6,8,5 ms. mean = 1412.4 ms. deviation = 2812.3001689008947 ms.

Test Suites: 1 passed, 1 total
Tests:       9 passed, 9 total
Snapshots:   0 total
Time:        7.714s, estimated 11s

Reply:
if we're just measuring Z profile performance, then the file opening etc is not a required performance metric for this test, and should be in a different test